### PR TITLE
Added tests for feature selection component

### PR DESCRIPTION
**Pull request by deutranium**
Added test for the `feature_selection` component.

- [x] Check artifact type for `feature_selection` artifact
- [x] Check artifact type for `updated_data` artifact

### CODE REVIEW

File: tfx_addons/feature_selection/component_test.py [hunk 1]
```diff
@@ -12,38 +12,54 @@
...
+
+from tfx_addons.feature_selection import component
+
+# getting the dataset
+_data_root = tempfile.mkdtemp(prefix='tfx-data')
+DATA_PATH = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv'
+_data_filepath = os.path.join(_data_root, "data.csv")
+urllib.request.urlretrieve(DATA_PATH, _data_filepath)
+
+context = InteractiveContext()
```
Comment (casassg) [inline tfx_addons/feature_selection/component_test.py:35]: InteractiveContext only works in Notebooks!

Comment (deutranium) [reply tfx_addons/feature_selection/component_test.py:35]: Oh okay, my bad. Would you have any resources for an alternate way? I wasn't able to find it as the examples I came across were implemented in Colab

@casassg

Comment (casassg) [reply tfx_addons/feature_selection/component_test.py:35]: Use LocalDagRunner instead. xgboost and sklearn examples have a test like that

**Comment by deutranium** [comment]
@casassg would this way work? Referred to xgboost and sklearn examples for the same 
Thanks!

File: tfx_addons/feature_selection/component_test.py [hunk 2]
```diff
@@ -12,38 +12,100 @@
...
+      self.assertEqual(2, len(m.store.get_artifacts_by_type("Examples")))
+      self.assertGreaterEqual(artifact_count, execution_count)
+      self.assertEqual(expected_execution_count, execution_count)
+
+    self.assertPipelineExecution()
 
 
 if __name__ == '__main__':
+  tf.compat.v1.enable_v2_behavior()
   tf.test.main()
```
Comment (casassg) [inline tfx_addons/feature_selection/component_test.py:109]: nit: let's indent this

**Comment by casassg** [comment]
I think this looks good. Would be great if we could check the result of the component to make sure it's selecting as expected but not sure if that's tested elsewhere

**Review by casassg — APPROVED** [review]

**Comment by casassg** [comment]
(also note that your tests are red see the test below)

**Comment by deutranium** [comment]
Thank you for your comments @casassg !
> Would be great if we could check the result of the component to make sure it's selecting as expected but not sure if that's tested elsewhere

Yup, just added a test to check correctness of selected features and it seems to be running fine (locally at least)

> (also note that your tests are red see the test below)

1. I think the automerge one should be fixed once all the codeowners approve the PR
2. I'm not so sure about the CI test though. The logs show `ModuleNotFoundError: No module named 'sklearn'` but how can I specify the same? I have tried adding it to requirements.txt but it doesn't seem to fix the issue

**Comment by casassg** [comment]
You will need to modify tfx_addons/version.py _PKG_METADATA dictionary to make sure to include what dependencies are needed for the component (key corresponds to project name, aka tfx_addons/{PROJECT_NAME}

**Review by Nirzu97 — APPROVED** [review]

File: tfx_addons/feature_selection/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+scikit_learn==1.1.2
+tensorflow
+tfx
+tfx_bsl==1.9.0
```
Comment (casassg) [inline tfx_addons/feature_selection/requirements.txt:4]: you can actually remove this from here as we should instead `pip install -e ".[feature_selection]"` (feel free to document that)

**Review by casassg — APPROVED** [review]

**Comment by casassg** [comment]
marking as ready to /merge you will need to fix the formatting issue on pre-commit check. Once fixed the PR should merge automatically

**Review by FatimahAdwan — APPROVED** [review]

File: tfx_addons/version.py [hunk 4]
```diff
@@ -56,6 +56,8 @@
     ],
     "feature_selection": [
         f"tfx{_TFXVERSION_CONSTRAINT}",
+        "scikit_learn>=1.0.2",
+        "tfx_bsl>=1.0.0"
     ],
```
Comment (casassg) [inline tfx_addons/version.py:61]:
```suggestion
    "feature_selection": 
    [f"tfx{_TFXVERSION_CONSTRAINT}", "scikit_learn>=1.0.2", "tfx_bsl>=1.0.0"],
```

File: tfx_addons/version.py [hunk 5]
```diff
@@ -54,9 +54,8 @@
     "schema_curation": [
         f"tfx{_TFXVERSION_CONSTRAINT}",
     ],
-    "feature_selection": [
-        f"tfx{_TFXVERSION_CONSTRAINT}",
-    ],
+    "feature_selection":
```
Comment (casassg) [inline tfx_addons/version.py:57]:
```suggestion
    "feature_selection":
```

**Pull request merged**